src/utils/__tests__/use-stable-prop.test.tsx

```diff
@@ -0,0 +1,132 @@
+import {useEffect} from 'react';
+import {renderHook, act} from '@testing-library/react-hooks';
+import {useStableProp} from '../use-stable-prop';
+
+type TestProp = {a?: number; b?: number} | undefined;
+
+describe('useStableProp', () => {
+  it('should trigger useEffect only when prop is not deeply equal', () => {
+    let prop: TestProp = {a: 1};
+    const effectSpy = jest.fn();
+
+    const useTestHook = (value: TestProp) => {
+      const stableValue = useStableProp(value);
+      useEffect(() => {
+        effectSpy();
+      }, [stableValue]);
+      return stableValue;
+    };
```

Comment:
I think this can be tested quite more simply, not introducing the complexity of using a `useEffect` in the test. We can rather test the core of the hook, that it should return the last stable reference of the passed parameter.

I think something like this is more to the core, and also easier to read:
```tsx
    const prop1 = {a: 1};
    const prop2 = {a: 1};

    const {result, rerender} = renderHook((prop) => useStableProp(prop), {
      initialProps: prop1,
    });

    expect(result.current).toBe(prop1);

    rerender(prop2);
    expect(result.current).toBe(prop1);
    expect(result.current).not.toBe(prop2);
```